code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -107,57 +107,73 @@
 		return
 	return ..()
 
-/obj/item/weapon/melee/baton/apply_hit_effect(mob/living/target, mob/living/user, var/hit_zone)
```

Comment:
Nanako is actually kinda right about this.

`var/target_zone = check_zone(user.zone_sel.selecting)` can be replaced with `var/target_zone = check_zone(hit_zone)`, if you leave `var/hit_zone` in the proc definition. `user.zone_sel_selecting` is called in the initial call to `attack()` anyways.